src/community/wmts-styles/src/main/java/org/geoserver/gwc/wmts/StylesExtension.java

```diff
@@ -0,0 +1,222 @@
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.LayerGroupInfo;
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.catalog.PublishedInfo;
+import org.geoserver.catalog.SLDHandler;
+import org.geoserver.catalog.StyleHandler;
+import org.geoserver.catalog.StyleInfo;
+import org.geoserver.catalog.Styles;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.gwc.layer.CatalogConfiguration;
+import org.geoserver.gwc.layer.GeoServerTileLayer;
+import org.geoserver.ows.Dispatcher;
+import org.geoserver.ows.LocalWorkspace;
+import org.geoserver.ows.Request;
+import org.geoserver.ows.URLMangler;
+import org.geoserver.ows.util.ResponseUtils;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.platform.ServiceException;
+import org.geotools.util.logging.Logging;
+import org.geowebcache.GeoWebCacheDispatcher;
+import org.geowebcache.conveyor.Conveyor;
+import org.geowebcache.io.XMLBuilder;
+import org.geowebcache.layer.TileLayer;
+import org.geowebcache.service.HttpErrorCodeException;
+import org.geowebcache.service.wmts.WMTSExtensionImpl;
+import org.geowebcache.storage.StorageBroker;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/** WMTS extension adding support for */
+public final class StylesExtension extends WMTSExtensionImpl implements ApplicationContextAware {
+
+    private static final Logger LOGGER = Logging.getLogger(StylesExtension.class);
+
+    private final Catalog catalog;
+
+    private List<ResourceFactory> resourceFactories;
+
+    public StylesExtension(
+            Catalog catalog, Dispatcher gsDispatcher, GeoWebCacheDispatcher gwcDispatcher) {
+        this.catalog = catalog;
+
+        // Allow dispatchers to handle put and delete too
+        // This should be done higher level, but it would require adding true support for
+        // these methods in the associated dispatcher, which is going to require significant work,
+        // at least in the case of the GeoServer one
+        gsDispatcher.setSupportedMethods("GET", "POST", "DELETE", "PUT", "OPTIONS");
+        gwcDispatcher.setSupportedMethods("GET", "POST", "DELETE", "PUT", "OPTIONS");
+    }
+
+    private final List<OperationMetadata> extraOperations = new ArrayList<>();
+
+    // {
+    //        extraOperations.add(new OperationMetadata("DescribeDomains"));
+    //        extraOperations.add(new OperationMetadata("GetDomainValues"));
+    //        extraOperations.add(new OperationMetadata("GetFeature"));
+    //        extraOperations.add(new OperationMetadata("GetHistogram"));
+    // }
```

Comment:
Is this intentionally left commented out?

Reply:
Good catch, forgot to remove it